editor/project_manager.cpp

```diff
@@ -454,6 +460,18 @@ class ProjectDialog : public ConfirmationDialog {
 
 			} else {
 				if (mode == MODE_NEW) {
+					// Before we create a project, check that the target folder is empty.
+					// If not, we need to ask the user if they're sure they want to do this.
+					if (!is_folder_empty) {
+						ConfirmationDialog *cd = memnew(ConfirmationDialog);
+						cd->set_title(TTR("Warning: This folder is not empty."));
+						cd->set_text(TTR("You are about to create a Godot project in a non-empty folder. Are you sure you wish to continue?"));
```

Comment:
In general, I'm OK with this, but I would also add more detailed warning, something like "Entire contents of this folder will be imported as project resources!", since this is the main concern with creating projects in non-empty folder.

Reply:
Does this look good?

![Screenshot from 2020-10-03 02-15-26](https://user-images.githubusercontent.com/1646875/94984851-aa583f80-051e-11eb-8c11-a50ee3b4c3a3.png)

Reply:
Yes, I think it's better. Maybe add extra empty line before `Are you sure...` to separate it?